lib/Analysis/NoiseAnalysis/Noise.h

```diff
@@ -40,13 +41,25 @@ class NoiseState {
   static NoiseState of(double value) {
```

Comment:
Is there a reason to keep `NoiseState::of` anymore? If not, it may be worth marking it as deprecated so the compiler can point out where to migrate any remaining uses. If so, we should figure out a way to remove the need for it, as having a class with multiple semantics for its underlying storage is risky.

Reply:
Removed the original `NoiseState::of` with normal scale semantic and renamed `NoiseState::ofLog2` with log scale semantic to `::of`. Now the user should always expect to manipulate it in normal scale while its storage is in log scale. The only time the user becomes aware it is log-scale is when they call `getValue`, which is now carefully handled with comments saying it is in log-scale.